step-controller-plugins/step-controller-plugins-timeseries/src/main/java/step/plugins/timeseries/TimeSeriesService.java

```diff
@@ -103,32 +100,46 @@ private TimeSeriesAggregationQuery mapToQuery(FetchBucketsRequest request) {
-            // Check if a time series already exist for this execution
-            TimeSeriesAggregationResponse aggregationResponse = aggregationPipeline.newQuery().range(firstMeasurement.getBegin(), lastMeasurement.getBegin())
-                    .filter(Map.of(MeasurementPlugin.ATTRIBUTE_EXECUTION_ID, executionId)).split(1l).run();
-            long seriesCount = aggregationResponse.getMatrix().size();
-            if (seriesCount > 0) {
-                throw new ControllerServiceException("Time series already exist for this execution. Unable to rebuild it");
-            } else {
+        if (this.timeSeriesExists(executionId)) {
+            throw new ControllerServiceException("Time series already exist for this execution. Unable to rebuild it");
+        } else {
+            // we need to check if measurements exists
+            Equals measurementFilter = Filters.equals(MeasurementPlugin.ATTRIBUTE_EXECUTION_ID, executionId);
+            Measurement firstMeasurement = measurementCollection.find(measurementFilter,
+                    new SearchOrder(MeasurementPlugin.BEGIN, 1), 0, 1, 0).findFirst().orElse(null);
+            Measurement lastMeasurement = measurementCollection.find(measurementFilter,
+                    new SearchOrder(MeasurementPlugin.BEGIN, -1), 0, 1, 0).findFirst().orElse(null);
+            if (firstMeasurement != null && lastMeasurement != null) {
                 return asyncTaskManager.scheduleAsyncTask(t -> {
                     TimeSeriesBucketingHandler timeSeriesBucketingHandler = new TimeSeriesBucketingHandler(ingestionPipeline);
                     LongAdder count = new LongAdder();
+                    SearchOrder searchOrder = new SearchOrder("begin", 1);
                     // Iterate over each measurement and ingest it again
-                    measurementCollection.find(measurementFilter, null, null, null, 0).forEach(measurement -> {
+                    measurementCollection.find(measurementFilter, searchOrder, null, null, 0).forEach(measurement -> {
                         count.increment();
-                        timeSeriesBucketingHandler.processMeasurement(measurement);
+                        timeSeriesBucketingHandler.ingestExistingMeasurement(measurement);
                     });
                     return new TimeSeriesRebuildResponse(count.longValue());
                 });
+            } else {
+                throw new ControllerServiceException("No measurement found matching this execution id");
             }
-        } else {
-            throw new ControllerServiceException("No measurement found matching this execution id");
         }
     }
+
+    @Operation(operationId = "checkTimeSeries", description = "Check if the time-series was created for a specific execution")
+    @Secured(right = "execution-read")
+    @GET
+    @Path("/execution/{executionId}/exists")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
```

Comment:
Is the return type boolean compatible with @Produces(MediaType.APPLICATION_JSON)?